ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,630 @@
+        ):
+            # Create storage profiles if not available
+            is_storageprofile_available = self.storage_profile_obj.is_exist(
+                resource_name="ssd-storageprofile"
+            )
+            if not is_storageprofile_available:
+                storage_profile_data = templating.load_yaml(
+                    constants.STORAGE_PROFILE_YAML
+                )
+                templating.dump_data_to_temp_yaml(
+                    storage_profile_data, constants.STORAGE_PROFILE_YAML
+                )
+                self.ocp_obj.exec_oc_cmd(f"apply -f {constants.STORAGE_PROFILE_YAML}")
+
+        # Create storage cluster if not present already
+        is_storagecluster = self.storage_cluster_obj.is_exist(
+            resource_name=constants.DEFAULT_STORAGE_CLUSTER
+        )
+        if not is_storagecluster:
+            if (
+                self.ocs_version < version.VERSION_4_16
+                and self.ocs_version >= version.VERSION_4_14
+            ):
+                storage_cluster_data = templating.load_yaml(
+                    constants.OCS_STORAGE_CLUSTER_YAML
+                )
+                templating.dump_data_to_temp_yaml(
+                    storage_cluster_data, constants.OCS_STORAGE_CLUSTER_YAML
+                )
+                self.ocp_obj.exec_oc_cmd(
+                    f"apply -f {constants.OCS_STORAGE_CLUSTER_YAML}"
+                )
+            else:
+                storage_cluster_data = templating.load_yaml(
+                    constants.OCS_STORAGE_CLUSTER_UPDATED_YAML
+                )
+                templating.dump_data_to_temp_yaml(
+                    storage_cluster_data, constants.OCS_STORAGE_CLUSTER_UPDATED_YAML
+                )
+                self.ocp_obj.exec_oc_cmd(
+                    f"apply -f {constants.OCS_STORAGE_CLUSTER_UPDATED_YAML}"
+                )
+
+        # Creating toolbox pod
+        setup_ceph_toolbox()
+
+        # Check ux server pod, ocs-provider server pod and rgw pods are up and running
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.UX_BACKEND_SERVER_LABEL,
+            resource_count=1,
+            timeout=180,
+        )
+        # Native storageclients are created as part of ODF 4.16 subscription and each of rbd and
+        # cephfs storageclaims gets created automatically with the storageclient creation
+        if self.ocs_version >= version.VERSION_4_16:
+            # Validate native client is created in openshift-storage namespace
+            self.deployment.wait_for_csv(
+                self.ocs_client_operator, constants.OPENSHIFT_STORAGE_NAMESPACE
+            )
+            log.info(
+                f"Sleeping for 30 seconds after {self.ocs_client_operator} created"
+            )
+
+            # Validate storageclaims are Ready and associated storageclasses are created
+            verify_storageclient()
```

Comment:
It is a good practice to separate installation and complete verification

Reply:
+1 
all assertion methods, full "if-else" block need to run from 'verification' method

Reply:
Have separated verification method.